client/my-sites/hosting/site-admin-interface-card/use-select-interface-mutation.ts

```diff
@@ -1,14 +1,14 @@
-import { useMutation, UseMutationOptions, useQueryClient } from '@tanstack/react-query';
-import { useCallback } from 'react';
+import { useMutation, UseMutationOptions } from '@tanstack/react-query';
 import wp from 'calypso/lib/wp';
-import { useDispatch } from 'calypso/state';
+import { useDispatch, useSelector } from 'calypso/state';
 import { requestAdminMenu } from 'calypso/state/admin-menu/actions';
-import { requestSite } from 'calypso/state/sites/actions';
+import getRawSite from 'calypso/state/selectors/get-raw-site';
+import { receiveSite, requestSite } from 'calypso/state/sites/actions';
 
 const SET_SITE_INTERFACE_MUTATION_KEY = 'set-site-interface-mutation-key';
 
 interface MutationResponse {
-	message: string;
+	WPCOM_ADMIN_INTERFACE: 'wp-admin' | 'calypso';
```

Comment:
I know that this works because we tested it together already but I have a question why changing it from string to literal type works in this case?

Reply:
The mutation response was outdated.
The key changed from `message` to `WPCOM_ADMIN_INTERFACE`. This was a bug on types.
And I decided to be more explicit here and add the values that actually returns `wp-admin` and `calypso`. This gives more information than just string.
All this wasn't needed until now, because we weren't actually reading the data that the mutation API returned.